### PR TITLE
Task: Changing userName and password & Report Summary Title

### PR DESCRIPTION
**Pull request by luankirkwood**

### CODE REVIEW

File: AgentDiscoveries-Backend/src/main/java/org/softwire/training/api/routes/v1/EntityCRUDRoutes.java [hunk 1]
```diff
@@ -1,16 +1,14 @@
 package org.softwire.training.api.routes.v1;
 
-import org.softwire.training.api.models.FailedRequestException;
```
Comment (JackMead) [inline AgentDiscoveries-Backend/src/main/java/org/softwire/training/api/routes/v1/EntityCRUDRoutes.java]: Thanks for cleaning these!

File: AgentDiscoveries-Backend/src/main/java/org/softwire/training/api/routes/v1/RegionSummaryReportsRoutes.java [hunk 2]
```diff
@@ -46,6 +42,8 @@ protected RegionSummaryReport validateThenMap(RegionSummaryReportApiModel apiMod
         model.setStatus(apiModel.getStatus());
         model.setReportTime(reportTimeUtc);
         model.setReportBody(apiModel.getReportBody());
+        model.setReportTitle(apiModel.getReportTitle());
+
```
Comment (JackMead) [inline AgentDiscoveries-Backend/src/main/java/org/softwire/training/api/routes/v1/RegionSummaryReportsRoutes.java]: Can we clear this spare line? 😛

File: AgentDiscoveries-Backend/src/main/java/org/softwire/training/api/routes/v1/UsersRoutes.java [hunk 3]
```diff
@@ -103,6 +103,32 @@ public UserApiModel updateEntity(Request req, Response res, int id) {
...
+            throw new FailedRequestException(ErrorCode.INVALID_INPUT, "userId cannot be found");
+        }
+
+        User oldUser = optionalUser.get();
+
+        User user = new User(
+                userApiModel.getUsername(),
+                Strings.isNullOrEmpty(userApiModel.getPassword())
+                        ? oldUser.getHashedPassword()
+                        : passwordHasher.hashPassword(userApiModel.getPassword()),
```
Comment (JackMead) [inline AgentDiscoveries-Backend/src/main/java/org/softwire/training/api/routes/v1/UsersRoutes.java]: Well remembered to hash!

File: AgentDiscoveries-Backend/src/main/java/org/softwire/training/api/routes/v1/UsersRoutes.java [hunk 4]
```diff
@@ -103,6 +103,32 @@ public UserApiModel updateEntity(Request req, Response res, int id) {
...
+        permissionsVerifier.verifyIsAdminOrRelevantUser(req, id);
+
+        Optional<User> optionalUser = usersDao.getUser(id);
+        if (!optionalUser.isPresent()) {
+            throw new FailedRequestException(ErrorCode.INVALID_INPUT, "userId cannot be found");
+        }
+
+        User oldUser = optionalUser.get();
+
+        User user = new User(
```
Comment (JackMead) [inline AgentDiscoveries-Backend/src/main/java/org/softwire/training/api/routes/v1/UsersRoutes.java]: Would this be slightly clearer without having to create a new User?

E.g. we could just directly update the old user:

``` 
User currentUser = optionalUser.get();
if(Strings.isNullOrEmpty(userApiModel.getPassword()){
  currentUser.setHashedPassword( //hash ... );
}

currentUser.setUsername(userApiModel.getUsername());

usersDao.updateUser(currentUser);
```

File: AgentDiscoveries-Frontend/app/src/components/profile/edit-profile-username-password.jsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+import Message from '../message';
+
+export default class EditProfileUsernamePassword extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            username: '',
+            password: '',
+
```
Comment (JackMead) [inline AgentDiscoveries-Frontend/app/src/components/profile/edit-profile-username-password.jsx]: Could we clear the spare lines here?

File: AgentDiscoveries-Frontend/app/src/components/profile/profile.jsx [hunk 6]
```diff
@@ -90,13 +98,6 @@ export default class Profile extends React.Component {
...
-    getCallSign() {
-        apiGet('agents', currentAgentId())
-            .then (agent=> {
-                this.setState({ agent: agent});
-                if(agent.agentId){
-                    this.getAgent(agent.agentId);
-                }
-            });
-    }
+
```
Comment (JackMead) [inline AgentDiscoveries-Frontend/app/src/components/profile/profile.jsx]: Spare lines 😛

**Comment by JackMead** [comment]
Added some comments from me - worth noting that it would have been much nicer (read: easier for me to review!) if we had separated the two changes

**Comment by luankirkwood** [comment]
I agree, I forgot to branch off before making the other changes 😣

File: AgentDiscoveries-Backend/src/main/java/org/softwire/training/api/routes/v1/UsersRoutes.java (hunk 4 above)
Comment (luankirkwood) [reply AgentDiscoveries-Backend/src/main/java/org/softwire/training/api/routes/v1/UsersRoutes.java]: not too sure how to implement this and if its necessary as we based it off of updateEntity.

**Pull request merged**